src/swap/swap_utils.cairo

```diff
@@ -86,19 +93,270 @@ struct SwapCache {
+        // for example, if 10 ETH is swapped in and there is a negative price impact
+        // only 9.995 ETH may be swapped in
+        // the remaining 0.005 ETH will be stored in the swap impact pool
+        price_impact_amount =
+            market_utils::apply_swap_impact_with_cap(
+                *params.data_store,
+                *params.event_emitter,
+                *_params.market.market_token,
+                *_params.token_in,
+                cache.token_in_price,
+                price_impact_usd
+            );
+        let price_impact_amount_felt252: felt252 = (-price_impact_amount).into();
+        if (fees.amount_after_fees <= price_impact_amount_felt252.try_into().unwrap()) {
+            SwapError::SWAP_PRICE_IMPACT_EXCEEDS_AMOUNT_IN(
+                fees.amount_after_fees, price_impact_amount
+            );
+        }
+        cache.amount_in = fees.amount_after_fees - price_impact_amount_felt252.try_into().unwrap();
+        cache.amount_out = cache.amount_in * cache.token_in_price.min / cache.token_out_price.max;
+        cache.pool_amount_out = cache.amount_out;
+    }
+
+    // the amountOut value includes the positive price impact amount
+    if (_params.receiver != _params.market.market_token) {
+        IBankDispatcher { contract_address: *_params.market.market_token }
+            .transfer_out(cache.token_out, *_params.receiver, cache.amount_out);
+    }
+
+    let mut delta_felt252: felt252 = (cache.amount_in + fees.fee_amount_for_pool).into();
+    market_utils::apply_delta_to_pool_amount(
+        *params.data_store,
+        params.event_emitter,
+        _params.market,
+        *_params.token_in,
+        delta_felt252.try_into().unwrap(),
+    );
+
+    // the poolAmountOut excludes the positive price impact amount
+    // as that is deducted from the swap impact pool instead
+    delta_felt252 = cache.pool_amount_out.into();
+    market_utils::apply_delta_to_pool_amount(
+        *params.data_store,
+        params.event_emitter,
+        _params.market,
+        cache.token_out,
+        -delta_felt252.try_into().unwrap(),
+    );
+
+    let prices = market_utils::MarketPrices {
+        index_token_price: (*params.oracle).get_primary_price(*_params.market.short_token),
+        long_token_price: if (_params.token_in == _params.market.long_token) {
+            cache.token_in_price
+        } else {
+            cache.token_out_price
+        },
+        short_token_price: if (_params.token_in == _params.market.short_token) {
+            cache.token_in_price
+        } else {
+            cache.token_out_price
+        },
+    };
+
+    market_utils::validate_pool_amount(params.data_store, _params.market, *_params.token_in);
+    market_utils::validata_reserve(
+        params.data_store, _params.market, @prices, cache.token_out == *_params.market.long_token
+    );
+    let (pnl_factor_type_for_longs, pnl_factor_type_for_shorts) = if (cache
```

Comment:
Is this equivalent to:

```
_params.tokenIn == _params.market.longToken ? Keys.MAX_PNL_FACTOR_FOR_DEPOSITS : Keys.MAX_PNL_FACTOR_FOR_WITHDRAWALS,
cache.tokenOut == _params.market.shortToken ? Keys.MAX_PNL_FACTOR_FOR_WITHDRAWALS : Keys.MAX_PNL_FACTOR_FOR_DEPOSITS
```

Reply:
Yeah I had supposed that second value will be the opposite. 
Just in case I came back with inline if return value